src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,127 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private T[] data = (T[]) new Object[0];
+    private int size = data.length;
```

Comment:
- You should init fields in constructor
- Field data have to be Object and after that you can't cast it